Switch to multiarch/alpine

jedahan opened the pull request:
This PR switches from ubuntu to alpine, and has COPY_DIR merge into root instead of create a subdirectory, and documents how to customize your images.

It needs testing on more than just my linux server and macOS laptop.

The good:
```zsh
~/ docker images pi-maker && docker images alpi-maker | tail -n1
REPOSITORY          TAG                 IMAGE ID            CREATED             SIZE
pi-maker            latest              5c46bb8c5817        21 hours ago        1.05GB
alpi-maker          latest              0bc01378bb21        6 minutes ago       55.7MB
```

Having pi-maker based off of multiarch I think means we also get rid of the binfmt and qemu install dependencies since we just use `docker run --rm --privileged multiarch/qemu-user-static:register --reset` but I'm not totally sure about that. This is pretty new territory for me.

Let me know your thoughts!

Commit:
lowercase readme

Commit:
add make flash

Commit:
lowercase dockerfile

Commit:
lowercase license

Commit:
create build directory

Commit:
move docker-entrypoint to entrypoint

Commit:
use docker for binfmt support

Commit:
document customization

Commit:
Merge COPY_DIR to /

Commit:
formatting

westlane commented:
I am trying out the new alpine code and look forward to the possibility of switching to this approach. I did have to replace "install" with "add" in the Dockerfile for apk, since "install" command was not recognized. After doing so I am able to properly build the docker image. However, the run script appears to stall out when trying to unpack the Arch Linux file. Have you experienced anything similar to this?

![script paused](https://user-images.githubusercontent.com/32886/47971087-b1892100-e05b-11e8-871d-09452b4293d9.png)

Commit:
apk add is the correct command

Commit:
sort packages

Commit:
Update example build instructions for raspbian

Commit:
remove unused DATE variable

Commit:
Update to latest raspbian build

Commit:
exit instead of self-kill

Commit:
whitespace formatting

Commit:
Add progress bar to tar extraction

Commit:
Set default arch linux size to 3G

jedahan commented:
So I changed around some of the one-liners in extracting which should fix the hanging in bash, and I added a progress bar to untarring so it doesn't seem like its hanging.

svg of it working on my machine:
[termtosvg_cyxpo_tv.svg.zip](https://github.com/lantern-works/pi-maker/files/2673455/termtosvg_cyxpo_tv.svg.zip)

jedahan commented on bin/disk-init:
```diff
@@ -23,7 +23,7 @@ if [[ ! $(ls "${loopdev}"*1) ]]; then
     echo "Please run script once more to fix."
     echo "------------------------------------------"
-    kill 0
+    exit
```
So one thing that worries me - does this exit keep a running container around? Whereas kill 0 does not?

jedahan replied:
Also is this missing loopback partition issue documented anywhere else?

Commit:
copy directory correctly

Commit:
first goal in makefile is always default goal

Commit:
go back to kill 0 and rerun for missing loopdev
